### PR TITLE
Update method to exclude leaderboard pages

### PR DESCRIPTION
**Pull request by rowanhogan**
Only the `donationsleaderboards/v1/leaderboard` endpoint supports excluding pages by param, and only by pageGuid. I recently added some filtering on the event leaderboard method by page slug, but I've beefed it up to include slug, guid, old ID and even tag value for tag leaderboards.

I've also added it to all of the possible fundraising leaderboard fetch methods.

### CODE REVIEW

**Comment by danhenderson** [comment]
Nice. We always get the question if it is guid/id/short name, and we never know the answer. Now the answer is whatever you want.

File: source/api/leaderboard/justgiving/index.js [hunk 1]
```diff
@@ -133,6 +130,27 @@ const filterLeaderboardResults = (results = [], isTeam) => {
...
+    'shortName',
+    'slug',
+    'tagValue'
+  ]
+
+  return results.filter(page =>
+    identifierKeys.reduce((current, key) => {
+      if (!page[key]) return current
+      return current ? pageIds.indexOf(page[key].toString()) < 0 : false
+    }, true)
```
Comment (danhenderson) [inline source/api/leaderboard/justgiving/index.js:150]: I'm trying to do this in my head, but if we think of a page that has both `pageGuid` and `pageShortName` attributes, and the user is filtering by `pageGuid`.

I think it's going to go through that reduce, set it to false when it matches on `pageGuid`, but then the next pass it is going to reset it back to true for `pageShortName`? Maybe?

Comment (danhenderson) [reply source/api/leaderboard/justgiving/index.js:150]: Ah no, it won't because it will only do that ternary if current is still false

Comment (rowanhogan) [reply source/api/leaderboard/justgiving/index.js:150]: Yeah I spent a long time on this one, and it's now a much terser version than earlier incarnations.

Basically, if `current` is ever `false`, it's all over, that page is goooooooone.

**Review by danhenderson — APPROVED** [review]

**Pull request merged**